lib/utils.pm

```diff
@@ -2374,11 +2375,21 @@ sub ensure_ca_certificates_suse_installed {
     if (script_run('rpm -qi ca-certificates-suse') == 1) {
         my $host_version = get_var("HOST_VERSION") ? 'HOST_VERSION' : 'VERSION';
         my $distversion = get_required_var($host_version) =~ s/-SP/_SP/r;    # 15 -> 15, 15-SP1 -> 15_SP1
-        zypper_call("ar --refresh http://download.suse.de/ibs/SUSE:/CA/SLE_$distversion/SUSE:CA.repo");
-        if (is_sle_micro) {
-            transactional::trup_call('--continue pkg install ca-certificates-suse');
+        my $res = zypper_call("ar --refresh http://download.suse.de/ibs/SUSE:/CA/SLE_$distversion/SUSE:CA.repo", retry => 3, exitcode => [0, 1, 2, 3, 4]);
+        if ($res == 0) {
+            if (is_sle_micro) {
+                transactional::trup_call('--continue pkg install ca-certificates-suse');
+            } else {
+                zypper_call("in ca-certificates-suse");
+            }
         } else {
-            zypper_call("in ca-certificates-suse");
+            # Get SUSE SSL certificates from internal CA
+            script_retry('curl -k https://ca.suse.de/certificates/ca/SUSE_Trust_Root.crt -o /etc/pki/trust/anchors/SUSE_Trust_Root.crt', timeout => 100, delay => 30, retry => 5);
+            if (is_sle_micro) {
+                transactional::trup_call('--continue run update-ca-certificates -v');
+            } else {
+                my $res = script_run 'update-ca-certificates -v';
```

Comment:
why not `assert_script_run`?

Reply:
ok

Reply:
done

Reply:
Now it is `$res` + `assert_script_run` which dont play well together. My point in general was that passing the exit code might not be beneficial

Reply:
I agree and this topic is discussed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20383#discussion_r1799012287: depending on the final solution, script_run with ouptut or assert_... will be used accordingly.